torch/_dynamo/utils.py

```diff
@@ -1604,9 +1604,9 @@ def nnmodule_has_hooks(
 def to_numpy_helper(value):
     """Convert tensor and torch_np.ndarray to numpy.ndarray."""
     if isinstance(value, torch_np.ndarray):
-        return value.tensor.numpy()
+        return to_numpy_helper(value.tensor)
     elif isinstance(value, torch.Tensor):
-        return value.numpy()
```

Comment:
should this instead be `value.numpy(force = True)`? this would convert to cpu and do detach as well

Reply:
I didn't want to do the detach, as we have not played around with autograd and this new feature just yet, so leaving it as it is right now. Will keep this in mind in the future though.